tsdb/chunks/head_chunks.go

```diff
@@ -226,9 +231,53 @@ func listChunkFiles(dir string) (map[int]string, error) {
 		}
 		res[int(seq)] = filepath.Join(dir, fi.Name())
 	}
+
 	return res, nil
 }
 
+// repairLastChunkFile deletes the last file if it's empty.
+// Because we don't fsync when creating these file, we could end
+// up with an empty file at the end during an abrupt shutdown.
+func repairLastChunkFile(files map[int]string) (_ map[int]string, returnErr error) {
+	lastFile := -1
+	for seq := range files {
+		if seq > lastFile {
+			lastFile = seq
+		}
+	}
+
+	if lastFile <= 0 {
+		return files, nil
+	}
+
+	f, err := os.Open(files[lastFile])
```

Comment:
Instead of opening the file just to read the size, what do you think about directly calling `os.Stat(files[lastFile])`? Could simplify the code.

Reply:
:facepalm: I did not know about this method, thanks for the pointer!